File: native/gameplugin/android/IETSystemUtil.cpp
//
//  IETSystemUtil.cpp
//  blackjack
//
//  Created by geekgy on 8/24/16.
//
//

#include "IETSystemUtil.h"

long IETSystemUtil::getCpuTime()
{
    return 0;
}
std::string IETSystemUtil::getConfigValue(std::string key)
{
    return "";
}
std::string IETSystemUtil::getBundleId()
{
    return "";
}
std::string IETSystemUtil::getAppName()
{
    return "";
}
int IETSystemUtil::getAppVersion()
{
    return 1;
}
std::string IETSystemUtil::getCountryCode()
{
    return "";
}
std::string IETSystemUtil::getLanguageCode()
{
    return "";
}
std::string IETSystemUtil::getDeviceName()
{
    return "";
}
std::string IETSystemUtil::getSystemVersion()
{
    return "";
}
std::string IETSystemUtil::getNetworkState()
{
    return "";
}
void IETSystemUtil::showAlertDialog(std::string title, std::string message, std::string cancelBtnTitle, cocos2d::ValueVector otherBtnTitles, const std::function<void (int)> &func)
{}
void IETSystemUtil::showProgressDialog(std::string message, int percent)
{}
void IETSystemUtil::hideProgressDialog()
{}
void IETSystemUtil::showLoading(std::string message)
{}
void IETSystemUtil::hideLoading()
{}
void IETSystemUtil::showMessage(std::string message)
{}
void IETSystemUtil::vibrate()
{}
void IETSystemUtil::saveImage(std::string imgPath, std::string album, const std::function<void (bool, std::string)> &func)
{}
void IETSystemUtil::sendEmail(std::string subject, cocos2d::ValueVector, std::string emailBody, const std::function<void (bool, std::string)> &func)
{}
void IETSystemUtil::setNotificationState(bool enable)
{
}
void IETSystemUtil::postNotification(cocos2d::ValueMap map)
{}
void IETSystemUtil::share(cocos2d::ValueVector items)
{}
void IETSystemUtil::keychainSet(std::string key, std::string value)
{
}
std::string IETSystemUtil::keychainGet(std::string key)
{
    return "";
}
void IETSystemUtil::copyToPasteboard(std::string str)
{}


void IETSystemUtil::requestUrl(std::string requestType, std::string url, std::string data, const std::function<void (bool, std::string)> func)
{
    
    // HttpClient::getInstance()->setTimeoutForConnect(10);
    // HttpClient::getInstance()->setTimeoutForRead(10);
    // HttpRequest* request = new HttpRequest();
    // request->setUrl(url.c_str());
    
    // CCLOG("http data %s %s",url.c_str(),data.c_str());
    // if (strcmp(requestType.c_str(), "get") == 0) {
    //     request->setRequestType(HttpRequest::Type::GET);
    // } else if (strcmp(requestType.c_str(), "post") == 0) {
    //     request->setRequestType(HttpRequest::Type::POST);
    // } else {
    //     assert(false);
    // }
    // if (data.size() > 0) {

    //     request->setRequestData(data->c_str(), strlen(data->c_str()));
    // }
    // request->setResponseCallback([=](HttpClient *sender, HttpResponse *response){
    //     if (response == nullptr || !response->isSucceed())
    //     {
    //         func(false, "");
    //         return;
    //     }
        
    //     std::vector<char> *buffer = response->getResponseData();
        
    //     std::string bufffff(buffer->begin(),buffer->end());
    //     func(true, bufffff);
        
    //     CCLOG("response:%s",bufffff.c_str());
    // });
    // HttpClient::getInstance()->sendImmediate(request);
    // request->release();
    

}


std::string IETSystemUtil::getAppVersionName()
{
    
    return "";
}


int IETSystemUtil::getAppBuildNum()
{
    return 0;
}

File: native/gameplugin/scripting/lua-bindings/auto/api/IETSystemUtil.lua

--------------------------------
-- @module IETSystemUtil
-- @parent_module ccg

--------------------------------
-- 
-- @function [parent=#IETSystemUtil] showLoading 
-- @param self
-- @param #string message
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] postNotification 
-- @param self
-- @param #map_table map
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] openUrl 
-- @param self
-- @param #string url
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] hideProgressDialog 
-- @param self
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] urlDecode 
-- @param self
-- @param #string url
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] base64Decode 
-- @param self
-- @param #string input
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] share 
-- @param self
-- @param #array_table items
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getDeviceType 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] setNotificationState 
-- @param self
-- @param #bool enable
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] vibrate 
-- @param self
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getLanguageCode 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] keychainSet 
-- @param self
-- @param #string key
-- @param #string value
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getAppVersion 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getLocalCfgValue 
-- @param self
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getDeviceName 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getAppBuild 
-- @param self
-- @return int#int ret (return value: int)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getPlatformType 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] exitGame 
-- @param self
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] setBadgeNum 
-- @param self
-- @param #int num
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getNetworkState 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getIDFA 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] showProgressDialog 
-- @param self
-- @param #string message
-- @param #int percent
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] xxteaDecrypt 
-- @param self
-- @param #string input
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getCpuTime 
-- @param self
-- @return long#long ret (return value: long)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getCountryCode 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getAppBundleId 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] isJailbroken 
-- @param self
-- @return bool#bool ret (return value: bool)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getSystemName 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] xxteaEncrypt 
-- @param self
-- @param #string input
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getOnlineCfgValue 
-- @param self
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getPlatCfgValue 
-- @param self
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getDeviceModel 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] urlEncode 
-- @param self
-- @param #string url
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] keychainGet 
-- @param self
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getIDFV 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getUUID 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] uncompressZip 
-- @param self
-- @param #string file
-- @param #string path
-- @return bool#bool ret (return value: bool)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getSystemVersion 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] hideLoading 
-- @param self
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getDebugMode 
-- @param self
-- @return int#int ret (return value: int)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getCfgValue 
-- @param self
-- @param #string key
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] base64Encode 
-- @param self
-- @param #string input
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getCurrentTimeMills 
-- @param self
-- @return long long#long long ret (return value: long long)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] showMessage 
-- @param self
-- @param #string message
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getAppName 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] copyToPasteboard 
-- @param self
-- @param #string str
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] setLocalConfig 
-- @param self
-- @param #string filePath
        
--------------------------------
-- 
-- @function [parent=#IETSystemUtil] getInstance 
-- @param self
-- @return IETSystemUtil#IETSystemUtil ret (return value: IETSystemUtil)
        
return nil
